pavelib/utils/stanford_i18n_helpers.py

```diff
@@ -0,0 +1,137 @@
+"""
+Tools for stanford i18n tasks
+"""
+
+from datetime import datetime
+import copy
+import fnmatch
+import logging
+
+import polib
+
+
+LOG = logging.getLogger(__file__)
+
+def segment_pofile_lazy(filename, segments):
+    """Segment a .po file using patterns in `segments`.
+
+    The .po file at `filename` is read, and the occurrence locations of its
+    messages are examined.  `segments` is a list of single key-value pairs:
+    the keys are segment .po filenames, the values are lists of patterns::
+
+        [
+            {'django-studio.po': [
+                'cms/*',
+                'some-other-studio-place/*',
+            ]},
+            {'django-weird.po': [
+                '*/weird_*.*',
+            ]},
+        ]
+
+    If ANY of a message's occurrences match the patterns for a segment, then that
+    message is written to the new segmented .po file (the first segment with
+    matching patterns).
+
+    Any message that matches no segments is written back to
+    the original file.
+
+    Arguments:
+        filename (path.path): a path object referring to the original .po file.
+        segments (list of dicts): specification of the segments to create.
+
+    Returns:
+        a set of path objects, all the segment files written.
+
+    """
+    reading_msg = "Reading {num} entries from {file}"
+    writing_msg = "Writing {num} entries to {file}"
+    source_po = polib.pofile(filename)
+    LOG.info(reading_msg.format(file=filename, num=len(source_po)))  # pylint: disable=logging-format-interpolation
```

Comment:
You don't need to format strings going into log statements; it can lazily evaluated.

```python
LOG.info(
    "Reading %s entries from %s",
    len(source_po),
    filename,
)
```